packages/commuter-server/src/resources/generateMapping.js

```diff
@@ -0,0 +1,90 @@
+const path = require("path"),
+  jsonfile = require("jsonfile"),
+  fetch = require("node-fetch");
+
+const nbformatV4Schema = "https://raw.githubusercontent.com/jupyter/nbformat/master/nbformat/v4/nbformat.v4.schema.json";
+
+const writeFile = schema =>
+  jsonfile.writeFileSync(
+    path.resolve(__dirname, "commuter.es.mapping.json"),
+    schema,
+    { spaces: 2 }
+  );
+
+const add_type_info = nbSchema => {
+  const { properties } = nbSchema.properties.metadata;
+  //codemirror_mode key needs a type, and nbformat specifies
+  //   "oneOf": [
+  //   {
+  //     "type": "string"
+  //   },
+  //   {
+  //     "type": "object"
+  //   }
+  // ]
+  properties.language_info.properties.codemirror_mode.type = "object";
+  // ES doesn't support array types. Just use string type
```

Comment:
@rgbkrk We might have to normalize it before we index the document. And, I couldn't find a way to tell ElasticSearch to do it.